core/src/main/scala/filodb.core/store/ChunkSink.scala

```diff
@@ -12,8 +12,17 @@ import monix.execution.Scheduler
 import monix.reactive.Observable
 
 import filodb.core._
+import filodb.core.metadata.Schemas
+import filodb.memory.format.UnsafeUtils
 
-case class PartKeyRecord(partKey: Array[Byte], startTime: Long, endTime: Long, hash: Option[Int])
+case class PartKeyRecord(partKey: Array[Byte], startTime: Long, endTime: Long, shard: Int)
+
+object PartKeyRecord {
+  def getBucket(partKey: Array[Byte], schemas: Schemas, numBuckets: Int): Int = {
+    val hash = schemas.part.binSchema.partitionHash(partKey, UnsafeUtils.arayOffset)
+    (hash & Int.MaxValue) % numBuckets
```

Comment:
Nit pick, Perhaps an assert on numBuckets to be non zero?

Reply:
This is unlikely to happen since we have default configs. Even if it happens, an IllegalArg is not any better than DivideByZero exception. I'll leave it as is.